packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/util.ts

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable-next-line import/no-extraneous-dependencies */
+import * as RedshiftData from 'aws-sdk/clients/redshiftdata';
+import { DatabaseQueryHandlerProps } from '../handler-props';
+
+const redshiftData = new RedshiftData();
+
+export type ClusterProps = Omit<DatabaseQueryHandlerProps, 'handler'>;
+
+export async function executeStatement(statement: string, clusterProps: ClusterProps): Promise<void> {
+  const executeStatementProps = {
+    ClusterIdentifier: clusterProps.clusterName,
+    Database: clusterProps.databaseName,
+    SecretArn: clusterProps.adminUserArn,
+    Sql: statement,
+  };
+  const executedStatement = await redshiftData.executeStatement(executeStatementProps).promise();
+  if (!executedStatement.Id) {
+    throw new Error('Service error: Statement execution did not return a statement ID');
+  }
+  await waitForStatementComplete(executedStatement.Id);
+}
+
+const waitTimeout = 100;
+async function waitForStatementComplete(statementId: string): Promise<void> {
+  await new Promise((resolve: (value: void) => void) => {
+    setTimeout(() => resolve(), waitTimeout);
+  });
+  const statement = await redshiftData.describeStatement({ Id: statementId }).promise();
+  if (statement.Status !== 'FINISHED' && statement.Status !== 'FAILED' && statement.Status !== 'ABORTED') {
+    return waitForStatementComplete(statementId);
+  } else if (statement.Status === 'FINISHED') {
+    return;
+  } else {
+    throw new Error(`Statement status was ${statement.Status}: ${statement.Error}`);
+  }
```

Comment:
Can you ensure that the error message is easy to read and propagates all the way to the CloudFormation stack failures?

(say for instance, if the table name already exists)

Reply:
```
    2/11 |7:19:30 PM | UPDATE_FAILED        | Custom::RedshiftDatabaseQuery               | User/Resource/Resource/Default (UserFDDCDD17) Received response status [FAILED] from custom resource. Message returned: Statement status was FAILED\
: ERROR: user "awscdkredshiftclusterdatabaseuserc17d5ebd" already exists                                                                                                                                                                       
                                                                                                                                                                                                                                               
Logs: /aws/lambda/aws-cdk-redshift-cluster--QueryRedshiftDatabase3de-uW42bMBt0DTj                                                                                                                                                              
                                                                                                                                                                                                                                               
    at waitForStatementComplete (/var/task/util.js:34:15)                                                                                                                                                                                      
    at processTicksAndRejections (internal/process/task_queues.js:95:5)                                                                                                                                                                        
    at async Object.executeStatement (/var/task/util.js:18:5)                                                                                                                                                                                  
    at async createUser (/var/task/user.js:35:5)                                                                                                                                                                                               
    at async updateUser (/var/task/user.js:40:9)                                                                                                                                                                                               
    at async handler (/var/task/user.js:21:9) (RequestId: 6ce5c1cc-a09d-440c-bffa-7356b5e8bfc0)                                                                                                                                                
        new CustomResource (/Users/chaimber/opensource/aws-cdk/packages/@aws-cdk/core/lib/custom-resource.js:30:25)                                                                                                                            
        \_ new DatabaseQuery (/Users/chaimber/opensource/aws-cdk/packages/@aws-cdk/aws-redshift/lib/private/database-query.js:40:25)                                                                                                           
        \_ new User (/Users/chaimber/opensource/aws-cdk/packages/@aws-cdk/aws-redshift/lib/user.js:51:25)                                                                                                                                      
        \_ Object.<anonymous> (/Users/chaimber/opensource/aws-cdk/packages/@aws-cdk/aws-redshift/test/integ.database.js:33:14)                                                                                                                 
        \_ Module._compile (node:internal/modules/cjs/loader:1109:14)                                                                                                                                                                          
        \_ Object.Module._extensions..js (node:internal/modules/cjs/loader:1138:10)                                                                                                                                                            
        \_ Module.load (node:internal/modules/cjs/loader:989:32)                                                                                                                                                                               
        \_ Function.Module._load (node:internal/modules/cjs/loader:829:14)                                                                                                                                                                     
        \_ Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)                                                                                                                                                  
        \_ node:internal/main/run_main_module:17:47                                                                                                                                                                                            
```